prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -924,6 +924,16 @@ class Hl7Serializer(val metadata: Metadata) : Logging {
         terser.set("/PATIENT_RESULT/ORDER_OBSERVATION/OBSERVATION/OBX-23-7", "XX")
     }
 
+    private fun setTruncationLimitWithEncoding(value: String, truncationLimit: Int): Int {
+        val regex = "[^&~|]".toRegex()
+        val match = regex.containsMatchIn(value)
+        return if (match) {
+            truncationLimit.minus(2)
```

Comment:
Can there be multiple escape characters?

Reply:
Yep, there can be multiple and this wouldn't account for them all. Good catch! I'll adjust